pkg/endpoints/dashboard.go

```diff
@@ -130,72 +106,173 @@ func IsOpenShift(r Resource, installedNamespace string) bool {
-			deploymentImage := deployment.Spec.Template.Spec.Containers[0].Image
-			if strings.Contains(deploymentImage, "openshift-pipeline/tektoncd-pipeline-controller") && strings.Contains(deploymentImage, ":") {
-				s := strings.SplitAfter(deploymentImage, ":")
-				if s[1] != "" {
-					version = s[1]
-				}
-			}
-		}
+		deploymentLabels := deployment.GetLabels()
 
-		// For master of Tekton Pipelines
 		if version == "" {
-			version = deploymentAnnotations["pipeline.tekton.dev/release"]
+			version = deploymentLabels[thingSearchingFor+".tekton.dev/release"]
 		}
 
-		// For Tekton Pipelines 0.10.0 + 0.10.1
 		if version == "" {
-			version = deploymentAnnotations["tekton.dev/release"]
+			version = deploymentLabels["version"]
 		}
 
-		// For Tekton Pipelines 0.9.0 - 0.9.2
+		// Installs through the OpenShift Operator displays version differently
+		// This deals with the OpenShift Operator install
 		if version == "" {
-			deploymentImage := deployment.Spec.Template.Spec.Containers[0].Image
-			if strings.Contains(deploymentImage, "pipeline/cmd/controller") && strings.Contains(deploymentImage, ":") && strings.Contains(deploymentImage, "@") {
-				s := strings.SplitAfter(deploymentImage, ":")
-				if strings.Contains(s[1], "@") {
-					t := strings.Split(s[1], "@")
-					version = t[0]
+			if namespace == "openshift-pipelines" {
+				deploymentImage := deployment.Spec.Template.Spec.Containers[0].Image
+				if strings.Contains(deploymentImage, "openshift-pipeline/tektoncd-"+thingSearchingFor+"-controller") && strings.Contains(deploymentImage, ":") {
+					s := strings.SplitAfter(deploymentImage, ":")
+					if s[1] != "" {
+						version = s[1]
+					}
+				}
+			}
+		}
+	}
+
+	if version == "" && thingSearchingFor == "pipelines" {
+		for _, deployment := range deployments.Items {
+			deploymentAnnotations := deployment.Spec.Template.GetAnnotations()
+
+			if namespace == "openshift-pipelines" {
+				deploymentImage := deployment.Spec.Template.Spec.Containers[0].Image
+				if strings.Contains(deploymentImage, "openshift-pipeline/tektoncd-pipeline-controller") && strings.Contains(deploymentImage, ":") {
+					s := strings.SplitAfter(deploymentImage, ":")
+					if s[1] != "" {
+						version = s[1]
+					}
+				}
+			}
+
+			// For master of Tekton Pipelines
+			if version == "" {
+				version = deploymentAnnotations["pipeline.tekton.dev/release"]
+			}
+
+			// For Tekton Pipelines 0.10.0 + 0.10.1
+			if version == "" {
+				version = deploymentAnnotations["tekton.dev/release"]
+			}
+
+			// For Tekton Pipelines 0.9.0 - 0.9.2
+			if version == "" {
+				deploymentImage := deployment.Spec.Template.Spec.Containers[0].Image
+				if strings.Contains(deploymentImage, "pipeline/cmd/controller") && strings.Contains(deploymentImage, ":") && strings.Contains(deploymentImage, "@") {
+					s := strings.SplitAfter(deploymentImage, ":")
+					if strings.Contains(s[1], "@") {
+						t := strings.Split(s[1], "@")
+						version = t[0]
+					}
```

Comment:
@a-roberts what pipeline version do we need to support for our next dashboard patch release (0.5.3)? 0.5.0 supports 0.10, are we expecting to still support older versions?

Reply:
Yeah, so the approach here should be supplementary for detecting versions and not break anything we used to pick up